providers/apollo/write_test.go

```diff
@@ -0,0 +1,203 @@
+package apollo
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/amp-labs/connectors"
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/test/utils/mockutils/mockcond"
+	"github.com/amp-labs/connectors/test/utils/mockutils/mockserver"
+	"github.com/amp-labs/connectors/test/utils/testroutines"
+	"github.com/amp-labs/connectors/test/utils/testutils"
+)
+
+// nolint
+func TestWrite(t *testing.T) { // nolint:funlen,gocognit,cyclop
+	t.Parallel()
+
+	unsupportedResponse := testutils.DataFromFile(t, "unsupported.json")
+	opportunityCreationResponse := testutils.DataFromFile(t, "opportunity-write.json")
+	updateDealsResponse := testutils.DataFromFile(t, "update-deals.json")
+
+	tests := []testroutines.Write{
+		{
+			Name:         "Object Name is required",
+			Server:       mockserver.Dummy(),
+			ExpectedErrs: []error{common.ErrMissingObjects},
+		},
+		{
+			Name:         "RecordData is required",
+			Input:        common.WriteParams{ObjectName: "leads"},
+			Server:       mockserver.Dummy(),
+			ExpectedErrs: []error{common.ErrMissingRecordData},
+		},
+
+		{
+			Name:  "Unsupported object",
+			Input: common.WriteParams{ObjectName: "arsenal", RecordData: "dummy"},
+			Server: mockserver.Conditional{
+				Setup: mockserver.ContentJSON(),
+				If:    mockcond.MethodPOST(),
+				Then:  mockserver.Response(http.StatusBadRequest, unsupportedResponse),
+			}.Server(),
+			ExpectedErrs: []error{common.NewHTTPStatusError(http.StatusBadRequest,
```

Comment:
```suggestion
ExpectedErrs: []error{
	common.ErrBadRequest,
	errors.New(unsupportedResponse), // nolint:goerr113
},
```
I would like to point you out to this method so you fully understand how `ExpectedErrs` is used by all tests:
https://github.com/amp-labs/connectors/blob/1c3cd51bb7a659752211e6aff629fa819e2e051b/test/utils/testutils/checks.go#L34-L52

To sum it up: it checks if composite error contains all errors you have listed. If it is an error created using errors.New() it looks for that string.